core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -284,6 +284,9 @@ public ParametersAction merge(@CheckForNull ParametersAction overrides) {
     }
 
     private Object readResolve() {
+        if (parameters == null) { // JENKINS-39495
```

Comment:
Ideally you should suppress the FindBugs annotation here, but we do not report medium-level issues right now